### ISSUE TITLE
Date parsing

### ISSUE THREAD

**Issue by RomeroDiver** [720kb/angular-datepicker#59]
Hi,
I was working on my project where I use Your datepicker, and after some struggle I found out that date formats are not working correctly. What I mean, although we can use any date format we like due to *date-format* attribute, the directive's internal functions have problems with this. Date constructor uses *Date.parse* internally. And parse function understands only **ISO 8601 format**(here is [link](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse))  The documentation says that other formats may be used, but the result is unpredictable. In my particular case, when I use the ISO format, the min/max filters work, but when I use other format(e.g."dd-MM-yyyy"), the filters do not work(both on Firefox and Chrome). To be honest, I have no good idea how to tackle this problem. 
If You come up with anything, please let me know - I'd really like to know how to solve this problem :)
Thanks for a very cool datepicker!

**Comment by 45kb** [comment]
@RomeroDiver  Thank you for spotting out this, yes it needs a double check.
As soon as it's possible for us, we will surely focus on this!

**Comment by kikocastro** [comment]
Same issue here with min/max and `dd-MM-yyyy`

**Comment by kevinchevallier** [comment]
Yes same issue while trying to use date formats and set `date-min-limit="{{datestart}}"` or `date-max-limit="{{dateend}}"` to use two pickers as a range picker. Otherwise works like a charm.
